src/System.IO.FileSystem.AccessControl/src/System/Security/AccessControl/DirectoryObjectSecurity.cs

```diff
@@ -349,7 +349,7 @@ private bool ModifyAccess(AccessControlModification modification, ObjectAccessRu
                         if (result == false)
                         {
                             Debug.Assert(false, "Invalid operation");
-                            throw new Exception();
+                            throw new System.Security.SecurityException(SR.InvalidOperation_AccessControl);
```

Comment:
Shouldn't this just assert and return false? (modified should be set to false as well) Throwing doesn't make sense given the documentation for both this class and DiscretionaryAcl.

Reply:
See the previous comments about these cases.